main.tf

```diff
@@ -1,3 +1,167 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+resource "ibm_is_security_group_rule" "default_vpc_rule" {
+  for_each  = local.all_rules_map
+  group     = var.security_group_id
+  direction = each.value.direction
+  remote    = each.value.remote
+
+  dynamic "tcp" {
```

Comment:
We should try to add more tests / examples exercising all the logic here in the dynamic blocks to ensure we do not have regression in the future.